migration-to-python-3-only.md

```diff
@@ -0,0 +1,183 @@
+of people, in particular: `ipykernel`, `traitlets`, and `ipython_genutils`. In
+order to provide LTS support for IPython, the IPython team will provide
+patches for the packages required by IPython 5.x if they
+affect the behavior of IPython. Otherwise our normal conventions apply, and only
+the current stable version of these packages will be supported.
+
+
+### Source distribution installation.
+
+
+The Python package manager (`pip`) does not respect the `requires-python`
+metadata field, and `setuptools` does not provide a way to set a value of this
+metadata field as far as we know. This implies that if the IPython core team release a version of
+IPython that is not compatible with Python 2, a user installing IPython
+by running `pip install ipython` under Python 2 will end up with a
+broken installation.
+
+There is an open issue on pip to check compatibility with Python when installing
+packages:
+
+    https://github.com/pypa/pip/issues/3182
+
+This will not be an issue before we actually release a source version
+of a package which is not Python 2 compatible, but we likely want to have
+this fixed in pip as soon as possible, so that users have a chance to upgrade to
+this version **before** upgrading IPython.
+
+There are alternative ways around that, like making `ipython` a metapackage which
+expresses a dependency on (say) `ipython_real`
+
+
+# Notebook & Python 3 support.
+
+The Jupyter Protocol is designed to be language agnostic, so the language the
+Notebook server (or any frontend) is written in does not have to match the
+language of code within a notebook.
+
+For the next+1 version of the Notebook server (6.0), we propose to require Python 3.
+
+The rationale being that the notebook application is one of those that can
+benefit the most from recent Python improvement, such as asyncio or function annotations.
+
+Unlike for IPython, we do not believe a LTS support is necessary, as a Python 3
+only version of the notebook application can perfectly run a Python 2 kernel.
+
+In order to allow a progressive transition to Python 3 for users and downstream
+integrators, we propose to release the first Python 3 only version of the
+notebook 6 months after the first release of the Python 3 only version of IPython.
+
+We propose to clearly communicate on the next release of the notebook (5.0), that
+one of the next versions will require a Python 3 environment.
+
+Timelines:
+    - Notebook 5.0 – Python 2 compatible – Summer/Fall 2016
+    - IPython 6.0 – Python 3 Only – Early 2017
+    - Notebook 6.0 (or 7.0) – Python 3 only – Mid 2017
+
+
+## Challenges
+
+As for IPython above; the notebook package need to be able to express that it
+requires Python 3. This is mitigated by people installing `jupyter`: we can
+make that conditionally depend on different version of `notebook`, depending on
+the version of Python.
+
+In particular for Anaconda, one path forward would be to bundle the
```

Comment:
@Carreau @takluyver is there any record of the conversation with the Anaconda team? I would like to take a look if possible. Thanks!